reactive/flannel.py

```diff
@@ -1,254 +1,218 @@
-    # this is likely to run first, save the TLS credentials
-    if is_state('etcd.tls.available'):
-        cert_path = '/etc/ssl/flannel'
-        etcd.save_client_credentials('{}/client-key.pem'.format(cert_path),
-                                     '{}/client-cert.pem'.format(cert_path),
-                                     '{}/client-ca.pem'.format(cert_path))
-
-    set_state('flannel.etcd.credentials.placed')
-
-
-@when_not('flannel.installed')
-def install_flannel():
-    ''' Unpack and install the binary release of flannel '''
-    hookenv.status_set('maintenance', 'Installing flannel.')
-    flannel_package = hookenv.resource_get('flannel')
-
-    if not flannel_package:
-        hookenv.status_set('blocked', 'Missing flannel resource.')
-        return
-
-    # Handle null resource publication, we check if its filesize < 1mb
-    filesize = os.stat(flannel_package).st_size
-    if filesize < 1000000:
-        hookenv.status_set('blocked', 'Incomplete flannel resource.')
-        return
-
-    charm_dir = hookenv.charm_dir()
-
-    # Unpack and install the flannel resource
-    _unpack_resource(flannel_package, '{0}/files/flannel'.format(charm_dir))
-    _install_files('./files/flannel/flanneld', '/usr/local/bin/flanneld')
-
-    set_state('flannel.installed')
-
-
-@when('flannel.installed', 'flannel.etcd.credentials.placed', 'etcd.available')
-@when_not('flannel.sdn.configured')
-def render_flannel_config(etcd):
-    ''' Consume the flannel information, and render the config files to
-    initialize flannel installation '''
-    hookenv.status_set('maintenance', 'Rendering systemd files.')
-    context = _build_context(etcd)
-    render('flannel.service', '/lib/systemd/system/flannel.service', context)
-    _initialize_etcd_with_data(etcd)
-    host.service_start('flannel')
-    set_state('flannel.sdn.configured')
-
-
-@when('flannel.installed')
-def set_flannel_version():
-    ''' Surface the currently deployed version of flannel to Juju '''
-    # flanneld -version
-    # v0.6.1
-    cmd = 'flanneld -version'
-    p = Popen(cmd, shell=True,
-              stdin=PIPE,
-              stdout=PIPE,
-              stderr=STDOUT,
-              close_fds=True)
-    version = p.stdout.read()
-    if version:
-        hookenv.application_version_set(version.split(b'v')[-1].rstrip())
-
-
-@when('flannel.sdn.configured', 'host.connected')
-@when_not('flannel.host.relayed')
-def relay_sdn_configuration(plugin_host):
-    ''' send the flannel interface configuration to the principal unit '''
-    try:
-        subnet, mtu = _ingest_network_config()
-        cidr = hookenv.config('cidr')
-        plugin_host.set_configuration(mtu, subnet, cidr)
-        set_state('flannel.host.relayed')
-        hookenv.status_set('active', 'Flannel subnet {0}'.format(subnet))
-    except TypeError:
-        hookenv.status_set('waiting', 'Flannel is starting up.')
+    status_set('waiting', 'Waiting for etcd relation.')
```

Comment:
Shouldn't this be blocked, as it will require an action by the end user? the .connected state only signifies that its related, not that the details are made available, which in turn would be the 'waiting' status.